### PR TITLE
[WEB-60] Print Date Range UI

### PR DESCRIPTION
**Pull request by thacherT1D**
Jira Ticket - https://tidepool.atlassian.net/browse/WEB-60

### CODE REVIEW

File: app/components/PrintDateRangeModal.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          ))}
+        </Flex>
+      </DialogContent>
+      <DialogContent style={{ height: 450 }}>
+        <Paragraph1>Or select a custom date range</Paragraph1>
+        <DateRangePicker
+          startDate={dates.startDate}
+          endDate={dates.endDate}
+          onDatesChange={dates => setDates(dates)}
+          isOutsideRange={day => (moment().diff(day) < 0)}
```
Comment (cbwebdevelopment) [inline app/components/PrintDateRangeModal.js:77]: Can probably set the minWidth and minHeight properties here, and only set the minHeight when the datepicker is open to match the Figma spec

```suggestion
          isOutsideRange={day => (moment().diff(day) < 0)}
          themeProps={{
            minWidth: '580px',
            minHeight: datePickerOpen ? '300px' : undefined,
          }}
```

Note: you'll need to implement `datePickerOpen` state for this to work

File: app/components/PrintDateRangeModal.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        />
+        {error && (
+          <Flex mt={4} className={inputClasses}>
+            <Caption>
+              Must Select at number of days or custom date range
+            </Caption>
+          </Flex>
```
Comment (cbwebdevelopment) [inline app/components/PrintDateRangeModal.js:84]: No need for wrapping `Flex` component here, and caption should be given error color
```suggestion
          <Caption mt={2} color="feedback.danger">
            Please select a date range
          </Caption>
```

File: stories/PrintDateRangeModal.stories.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  );
+};
+
+PrintDateRangeModalStory.story = {
+  name: 'Print Date Range Modal',
+  parameters: {
+    design: {
+      type: 'figma',
+      url:
+        'https://www.figma.com/file/iuXkrpuLTXExSnuPJE3Jtn/Tidepool-Design-System---Sprint-1?node-id=8%3A826',
```
Comment (cbwebdevelopment) [inline stories/PrintDateRangeModal.stories.js:54]: Figma link should be updated.  Pointing to the `Dialog` component spec currently

**Review by cbwebdevelopment — CHANGES_REQUESTED** [review]
It's a solid start.  Most of the logic is on point -- but we should probably make sure the dates set are fully-inclusive (i.e start of the day for the startDate, end of the day for the endDate) and respective of the user's timezone.

When possible, we should leverage the rebass styling props over the generic `style` prop.  I have a few other theming/markup suggestions within the comments.

As discussed, with you being off for a vacay, I'll pick up on this and implement my suggested changes, and then pass it off for a second round of PR reviews

File: app/components/PrintDateRangeModal.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    open,
+    disabled,
+    error,
+    required,
+  } = props;
+
+  const [dates, setDates] = useState({ startDate: null, endDate: null });
+
+  const getLastNDays = (startDate) => ({
+    startDate: moment().subtract(startDate, 'days'),
```
[the remark this answers is not in the document]
Comment (cbwebdevelopment) [reply app/components/PrintDateRangeModal.js]: Implemented

**Comment by cbwebdevelopment** [comment]
@krystophv I did the initial PR review for @thacherT1D on this, but since she's away for a couple of weeks, I implemented my suggested changes and marked them as resolved.

Just passing to you now for a final review.

**Review by krystophv — APPROVED** [review]
LGTM 🚢

**Review by cbwebdevelopment — APPROVED** [review]

**Pull request merged**